packages/cli/src/controllers/AuthController.ts

```diff
@@ -0,0 +1,172 @@
+import validator from 'validator';
+import { Get, Post, RestController } from '@/decorators';
+import { AuthError, BadRequestError, InternalServerError } from '@/ResponseHelper';
+import { compareHash, sanitizeUser } from '@/UserManagement/UserManagementHelper';
+import { issueCookie, resolveJwt } from '@/UserManagement/auth/jwt';
+import { AUTH_COOKIE_NAME } from '@/constants';
+import type { Request, Response } from 'express';
+import type { ILogger } from 'n8n-workflow';
+import type { User } from '@db/entities/User';
+import type { LoginRequest, UserRequest } from '@/requests';
+import type { PublicUser } from '@/UserManagement/Interfaces';
+import { In, Repository } from 'typeorm';
+import type { Config } from '@/config';
+import type { IInternalHooksClass } from '@/Interfaces';
+
+@RestController()
+export class AuthController {
+	constructor(
+		private config: Config,
+		private internalHooks: IInternalHooksClass,
+		private userRepository: Repository<User>,
+		private logger: ILogger,
+	) {}
+
+	/**
+	 * Log in a user.
+	 * Authless endpoint.
+	 */
+	@Post('/login')
```

Comment:
As tech debt in future it would be useful to improve searchability of endpoints in all their docstrings, e.g. `POST /login`.

Reply:
perhaps something like `* @route {POST /login}` on line 28?